src/DetailsView/reports/components/report-file-name-generator.ts

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+type DateProvider = () => Date;
+
+export type ReportFileNameGeneratorDeps = {
+    dateProvider: DateProvider;
+};
+
+export class ReportFileNameGenerator {
```

Comment:
>ReportFileNameGenerator [](start = 13, length = 23)

Discussed offline with mark that we already have a ReportNameGenerator with very similar functionality (but which seems to be dead code right now, that shows up as issues table state that never gets read). Mark's going to look into consolidating

Reply:
I consolidated.